### PR TITLE
Add forms and comments

### PR DESCRIPTION
**Pull request by gracekabaghe**
- Create forms to perform the following functions:
- Creates a Post on behalf of the current_user.
- Create a comment on behalf of the current_user.
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    if @new_comment.save
+      # @new_comment.count_comments(@post.id)
+      Comment.count_comments(@post.id)
+      flash[:success] = 'Your comment has been added!'
+      redirect_to user_post_path(@user, @post)
+    else
+      flash.now[:error] = 'Comment could not be added'
+      # render user_post_path(@user, @post)
+    end
+  end
```
Comment (Emmyn5600) [inline app/controllers/comments_controller.rb]: - [x] To keep your code clean and professional please remove the commented code.

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    #   flash[:alert] = 'You have already liked this post'
+    # else
+    @like = Like.create(user_id: user.id, post_id: user_post.id)
+    if @like.save
+      Like.count_likes(user_post.id)
+      flash[:success] = 'Your comment has been added!'
+
+    else
+      flash.now[:error] = 'Comment could not be added'
+    end
```
Comment (Emmyn5600) [inline app/controllers/likes_controller.rb]: - [x] To keep your code clean and professional please remove the commented code.
- [x] This method is for creating the likes not comment, please change the flash message.

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (Emmyn5600) [inline app/controllers/comments_controller.rb]: - [x] To keep your code clean and professional please remove the commented code.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,16 +1,47 @@
...
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @id = params[:user_id]
+    @user = User.find(@id)
+    @posts = User.all_post(@id)
   end
 
   def show
+    # @post = Post.find(params[:id])
```
Comment (Emmyn5600) [inline app/controllers/posts_controller.rb]: - [x] To keep your code clean and professional please remove the commented code.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @gracekabaghe 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:

## Required Changes :recycle:

- [ ] Check the comment under the review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Dipeshtwis — APPROVED** [review]
Hi @gracekabaghe, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**